iib/workers/tasks/build.py

```diff
@@ -807,6 +807,17 @@ def handle_add_request(
         ),
     )
     from_index_resolved = prebuild_info['from_index_resolved']
+
+    # FIXME: To be removed when the binaryless image build support is implemented
+    is_binaryless = prebuild_info['binary_image'] == "scratch"
+    if is_binaryless:
+        _cleanup()
+        log.warning("IIB is not yet able to process binaryless images.")
+        set_request_state(
+            request_id, 'complete', 'IIB is not yet able to process binaryless images'
```

Comment:
Is 'complete' the correct state here? I would prefer 'failed'.

Reply:
Ok!

I'll patch it accordingly as Tulsi and you agree on that